### DISCUSSION TITLE
How to add meta in route

### DISCUSSION THREAD

**Discussion by ibrahim-990** [hannoeru/vite-plugin-pages#397] [Q&A]
### Describe the bug

<route lang="yaml">
meta:
  permission:
    parent: roles
    action: !expr >
      ${isEdit.value ? 'can_update' : 'can_view'}
</route>
i cant access isEdit from <script>const isEdit = ref(false)</script> in route with vuejs.
i Get value like this
permission: {parent: 'roles', action: "${isEdit.value ? 'can_update' : 'can_view'}"}

### Reproduction

no repo

### System Info

```shell
Browsers
```

### Used Package Manager

npm

### Validations

- [X] Follow the [Code of Conduct](https://github.com/hannoeru/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the docs.
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A please open a GitHub Discussion.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by jbaubree** [comment]
https://github.com/hannoeru/vite-plugin-pages#sfc-custom-block-for-route-data

**Comment by ibrahim-990** [comment]
> https://github.com/hannoeru/vite-plugin-pages#sfc-custom-block-for-route-data

i cant access isEdit from script in <route>!!

**Comment by ibrahim-990** [comment]
@jbaubree

**Comment by jbaubree** [comment]
Can you share a repo with repro ?

**Comment by ibrahim-990** [comment]
I'm work on vuexy dashboard
![image](https://user-images.githubusercontent.com/65185082/231711060-fd9fafb3-6c98-4ed5-9bd1-149e40a497e3.png)
I'm use like this

**Comment by ibrahim-990** [comment]
@jbaubree

**Comment by ibrahim-990** [comment]
@jbaubree can you help me please ?

**Comment by jbaubree** [comment]
I don't understand why you wan't to use route for this. You have isEdit in script and you can build your input types and submit button depending on isEdit or not.
What do you want to do with action in route?

**Comment by jbaubree** [comment]
You can use route meta to define if you need specific role or auth to show this page for example:

// yourPage.vue
```
<route lang="yaml">
meta:
  needAuth: true
</route>
```
// router.ts
```
router.beforeEach(({ meta }) => {
    const userStore = useUserStore()
    if (meta.needAuth && !userStore.isAuthenticated))
      return { name: 'sign-in' }
    return true
  })
```

**Comment by ibrahim-990** [comment]
ok this work,
but i have on form page
and this form has meta permission edit or add
and i want to check like this to add permission ${isEdit.value ? 'can_update' : 'can_view'} 
how i can do this?

**Comment by jbaubree** [comment]
your edit or add state does not need to be in route meta, or i do not understand why.

Edit: you need to move this issue to discussion
